src/ast/nodes/CallExpression.ts

```diff
@@ -135,15 +136,15 @@ export default class CallExpression extends NodeBase {
 		});
 	}
 
-	reassignPath(path: ObjectPath, options: ExecutionPathOptions) {
-		!options.hasReturnExpressionBeenAssignedAtPath(path, this) &&
+	reassignPath(path: ObjectPath) {
+		if (path.length > 0 && !this.context.reassignmentTracker.track(this, path)) {
```

Comment:
So reassignment is basically a deoptimization, and the reassignment tracker allows us to track this globally? Or is it again about managing recursion here?

Reply:
Yes, reassignments are a global deoptimization but basically, it is still only about recursion tracking. The reason I had to add this here is in addition to the usages in `LocalVariable` is this situation:

```js
const foo = () => foo();
foo().bar = 'baz';
```

The second assigment will trigger a "forEachReturnExpression"  with a reassignment for `foo`. This again will try to reassign the return expression `foo()` where we loose the recursion tracker of "forEachReturnExpression" and then another cycle starts. To break this cycle, one way was to extend the active tracking to call expressions.